dll/win32/shell32/CShellDispatch.cpp

```diff
@@ -384,9 +519,34 @@ HRESULT STDMETHODCALLTYPE CShellDispatch::ExplorerPolicy(BSTR policy, VARIANT *v
     return E_NOTIMPL;
 }
 
+#ifndef SSF_SERVERADMINUI
+#define SSF_SERVERADMINUI 4
+#endif
 HRESULT STDMETHODCALLTYPE CShellDispatch::GetSetting(LONG setting, VARIANT_BOOL *result)
 {
     TRACE("(%p, %lu, %p)\n", this, setting, result);
+
+    int flag = -1;
+    SHELLSTATE ss = { };
+    SHGetSetSettings(&ss, setting, FALSE);
+    switch(setting)
+    {
+        case SSF_SHOWALLOBJECTS:   flag = ss.fShowAllObjects;     break;
+        case SSF_SHOWEXTENSIONS:   flag = ss.fShowExtensions;     break;
+        case SSF_SHOWSYSFILES:     flag = ss.fShowSysFiles;       break;
+        case SSF_DONTPRETTYPATH:   flag = ss.fDontPrettyPath;     break;
+        case SSF_NOCONFIRMRECYCLE: flag = ss.fNoConfirmRecycle;   break;
+        case SSF_SHOWSUPERHIDDEN:  flag = ss.fShowSuperHidden;    break;
+        case SSF_SEPPROCESS:       flag = ss.fSepProcess;         break;
+        case SSF_STARTPANELON:     flag = ss.fStartPanelOn;       break;
+        case SSF_SERVERADMINUI:    flag = IsOS(OS_SERVERADMINUI); break;
+    }
+    if (flag >= 0)
+    {
+        *result = flag ? VARIANT_TRUE : VARIANT_FALSE;
+        return S_OK;
+    }
+
     return E_NOTIMPL;
```

Comment:
Is it still a correct return value here? Maybe it should be changed to `E_FAIL` or something else instead? :thinking: Since this function is actually implemented now.